examples/fpga/artya7/top_artya7.core

```diff
@@ -43,6 +48,7 @@ targets:
     filesets:
       - files_rtl_artya7
       - files_constraints
+      - power_analysis ? (files_tcl)
```

Comment:
This is clever! But is it possible to run `vivado_setup_hooks.tcl` unconditionally and then check for the `power_analysis` parameter from there? Otherwise, we're going to struggle to add another mode in the future.

Reply:
Thanks for checking the code it really helps me! 

I am not sure how/if we can use the user defined FuseSoc flag inside `vivado_setup_hooks.tcl.tcl` file. Another solution would be defining a vlogparam and then use it in the tcl as you suggested. I am going to look into it further for sure.

Reply:
Ah: I hadn't twigged about the difference between a flag and a parameter. @imphil: Do you have an opinion on the best way to do this?

Reply:
I updated the script to do the deciding after sourcing. I think this looks cleaner and creates no worries about future additions.